controller_interface/include/controller_interface/chainable_controller_interface.hpp

```diff
@@ -57,10 +59,11 @@ class ChainableControllerInterface : public ControllerInterfaceBase
   bool is_chainable() const final;
 
   CONTROLLER_INTERFACE_PUBLIC
-  std::vector<hardware_interface::StateInterface> export_state_interfaces() final;
+  std::vector<std::shared_ptr<hardware_interface::StateInterface>> export_state_interfaces() final;
```

Comment:
```suggestion
  std::vector<std::shared_ptr<const hardware_interface::StateInterface>> export_state_interfaces() final;
```
I think it would make sense to use aliasing with `using`to have something like `hardware_interface::StateInterface::ConstSharedPtr` to `std::shared_ptr<const hardware_interface::StateInterface>`

Reply:
@mamueluth would this work with this `const`?

Reply:
That's a different PR now